fmn/fmn_fasshim.py

```diff
@@ -71,65 +91,92 @@ def make_fas_cache(**config):
     _cache.set('fas_cache_built', True)
 
 
-def update_nick(username):
-    global fasclient
-    try:
-        log.info("Downloading FAS cache for %s*" % username)
-        request = fasclient.send_request(
-            '/user/list',
-            req_params={'search': '%s' % username},
-            auth=True)
-    except fedora.client.ServerError as e:
-        log.warning(
-            "Failed to download fas cache for %s: %r" % (username, e))
-        return {}
-
-    log.info("Caching necessary data for %s" % username)
-    for user in request['people']:
-        nick = user['ircnick']
-        if nick:
+def _add_to_cache(users):
+    for user in users:
+        nicks = user.get('ircnicks', [])
+        for nick in nicks:
             _cache.set(nick, user['username'])
 
-        email = user['email']
-        if email:
+        emails = user.get('emails', [])
+        for email in emails:
             _cache.set(email, user['username'])
+
+
+def update_nick(username):
+    global client
+    if config.get('fasjson'):
+        try:
+            log.info("Downloading FASJSON cache for %s*" % username)
+            response = client.get_user(username=username)
+            _add_to_cache([response.json().result])
```

Comment:
Are you sure it's not `.json()["result"]` ? Also, I think you could move the `.json()` calls into the `Client` class.